pkg/scheduler/manager.go

```diff
@@ -168,13 +168,32 @@ func (mgr *PodGroupManager) calculateAssignedPods(target *corev1.Pod) int {
 	return count
 }
 
-func (mgr *PodGroupManager) getCreationTimestamp(pod *corev1.Pod, defaultTime time.Time) time.Time {
+// rescheduleAfterAll is a method to reschedule pod group to end of queue.
+// the duration is the delay of current last pog group.
+func (mgr *PodGroupManager) rescheduleAfterAll(podGroup *apiv1.PodGroup, dur time.Duration) (*apiv1.PodGroup, error) {
+	podGroups, err := mgr.schedulingClient.PodGroups(podGroup.Namespace).List(mgr.ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("fail to list pod group: %s", err.Error())
+	}
+
+	maxTime := time.Time{}
+	for _, pg := range podGroups.Items {
+		if pg.SchedulingTime().After(maxTime) {
+			maxTime = pg.SchedulingTime()
+		}
+	}
+
+	podGroup.Status.NextSchedulingTime.Time = maxTime.Add(dur)
```

Comment:
The name `SchedulingTime` isn't accurate since it's not scheduled for that time.

By the way, why not uses the now timestamp, If just for sorting in the queue, it's enough and simple to understand.

Reply:
Sounds reasonable.

Reply:
Fixed